### PR TITLE
Add graphics engine

### PR DESCRIPTION
**Pull request by benjxia**
Good luck reviewing this monstrosity. Will not merge until everyonre reviews.

### CODE REVIEW

File: lib/stb/data/map_01.png
Comment (Alanine42) [inline lib/stb/data/map_01.png:1]: Didn't expect to see Einstein here/

**Review by Alanine42 — APPROVED** [review]
i don't know what i'm looking at, approved

**Comment by benjxia** [comment]
> i don't know what i'm looking at, approved

bruh

**Review by AnnLe4869 — APPROVED** [review]
What on Earth I am looking at? Approve

**Comment by benjxia** [comment]
> What on Earth I am looking at? Approve

bruh

**Comment by benjxia** [comment]
pls test before u approve >:(((((((((((((

**Review by Alanine42 — APPROVED** [review]
looks like u figured something out, approved x2

**Review by realmatthewpeng — APPROVED** [review]
Works on my machine

**Comment by benjxia** [comment]
screw it, merging, rana and melody have already tested it

**Pull request merged**